frontends/uhdm/UhdmAst.cc

```diff
@@ -248,8 +248,55 @@ static void add_or_replace_child(AST::AstNode* parent, AST::AstNode* child) {
 			*it = child;
 			return;
 		}
+	} else if (child->type == AST::AST_INITIAL) {
+		// Special case for initials
+		// Ensure that there is only one AST_INITIAL in the design
+		// And there is only one AST_BLOCK inside that initial
+		// Copy nodes from child initial to parent initial
+		auto initial_node_it = find_if(parent->children.begin(), parent->children.end(),
+			[] (AST::AstNode *node) {return (node->type == AST::AST_INITIAL);} );
+		if (initial_node_it != parent->children.end()) {
+			AST::AstNode* initial_node = *initial_node_it;
+			auto block_node_it = find_if(initial_node->children.begin(), initial_node->children.end(),
```

Comment:
Isn't ``AST_BLOCK`` need to be 1st children of ``AST_INITIAL`` (and only children node of ``AST_INITIAL``)? 

If yes, we should just check, if there isn't any node in AST_INITIAL or there is and first children is ``AST_BLOCK`` (and fail if not).

Reply:
I didn't find any evidence in code that will prove that there can be other nodes than AST_BLOCK as AST_INITIAL children so I think that You are correct. I modified the code to check only if there are any children in AST_INITIAL and if the first child is AST_BLOCK. If those conditions aren't met we throw errors with `log_assert()`. This change greatly simplified the code, thank You for the idea!